pydis_core/utils/paste_service.py

```diff
@@ -0,0 +1,95 @@
+
+
+class PasteUploadError(Exception):
+    """Raised when an error is encountered uploading to the paste service."""
+
+
+class PasteTooLongError(Exception):
+    """Raised when content is too large to upload to the paste service."""
+
+
+async def send_to_paste_service(
+    *,
+    contents: str,
+    paste_url: str,
+    http_session: ClientSession,
+    file_name: str = "",
+    lexer: str = "python",
+    max_size: int = MAX_PASTE_SIZE,
+) -> str:
+    """
+    Upload some contents to the paste service.
+
+    Args:
+        contents: The content to upload to the paste service.
+        paste_url: The base url to the paste service.
+        http_session (aiohttp.ClientSession): The session to use when POSTing the content to the paste service.
+        file_name: The name of the file to save to the paste service.
+        lexer: The lexer to save the content with.
+        max_size: The maximum number of bytes to be allowed. Anything larger that 256kB will be rejected.
+
+    Raises:
+        :exc:`ValueError`: ``max_length`` greater than the maximum allowed by the paste service.
+        :exc:`PasteTooLongError`: ``contents`` too long to upload.
+        :exc:`PasteUploadError`: Uploading failed.
+
+    Returns:
+        The URL to the uploaded contents.
+    """
+    if max_size > MAX_PASTE_SIZE:
+        raise ValueError(f"`max_length` must not be greater than {MAX_PASTE_SIZE}")
+
+    contents_size = len(contents.encode())
+    if contents_size > max_size:
+        log.info("Contents too large to send to paste service.")
+        raise PasteTooLongError(f"Contents of size {contents_size} greater than maximum size {max_size}")
+
+    log.debug(f"Sending contents of size {contents_size} bytes to paste service.")
+    payload = {
+        "expiry": "1month",
+        "files": [
+            {"name": file_name, "lexer": lexer, "content": contents},
+        ]
+    }
+    for attempt in range(1, FAILED_REQUEST_ATTEMPTS + 1):
+        try:
+            async with http_session.post(f"{paste_url}/api/v1/paste", json=payload) as response:
+                response_json = await response.json()
+        except ClientConnectorError:
+            log.warning(
+                f"Failed to connect to paste service at url {paste_url}, "
+                f"trying again ({attempt}/{FAILED_REQUEST_ATTEMPTS})."
+            )
+            continue
+        except Exception:
+            log.exception(
+                f"An unexpected error has occurred during handling of the request, "
+                f"trying again ({attempt}/{FAILED_REQUEST_ATTEMPTS})."
+            )
+            continue
+
+        if response.status == 400:
+            log.warning(
+                f"Paste service returned error {response_json['message']} with status code {response.status}, "
+                f"trying again ({attempt}/{FAILED_REQUEST_ATTEMPTS})."
+            )
+            continue
+
+        if link := response_json.get("link" ):
```

Comment:
```suggestion
        if link := response_json.get("link"):
```
Do we not have linting here?

Reply:
We _should_ not sure why this wasn't picked up. I've updated this now anyway, to simply check for a 200 status code.